node/src/node_configurator/mod.rs

```diff
@@ -207,6 +217,35 @@ mod tests {
         )
     }
 
+    #[test]
+    fn data_directory_from_dir_opt_creates_new_folder_for_every_blockchain_platform() {
+        let dirs_wrapper = DirsWrapperMock::new()
+            .home_dir_result(Some(PathBuf::from("/nonexistent_home/root".to_string())))
+            .data_dir_result(Some(PathBuf::from("/nonexistent_home/root/.local/share")));
+        let real_user = RealUser::new(
+            None,
+            None,
+            None,
+        );
+        let custom_dir = Some(PathBuf::from("~/mynode".to_string())); 
+        let data_dir_opt = custom_dir;
+        let chain_name = "eth-ropsten";
```

Comment:
You might not be aware of that, so no stress, but the Ropsten testnet has been shut down. Avoid using that name, you should pick something like "eth-mainnet" or "polygon-mumbai". 

If I'm informed correctly Dan put some changes removing all references to Ropsten in his unfinished card and so you don't have to care too much about tests around the new one.

Reply:
thanks :)